Update table statistics description in Iceberg doc (#24084)
trinodb/trino: docs/src/main/sphinx/connector/iceberg.md

* - `iceberg.table-statistics-enabled`
  - Enable [](/optimizer/statistics). The equivalent [catalog session
    property](/sql/set-session) is `statistics_enabled` for session specific

>     use. Set to `false` to prevent statistics from being used by

Reviewer: ```suggestion
    use. Set to `false` to prevent statistics usage by the
```

    [](/optimizer/cost-based-optimizations) to make better decisions about the

>     query plan. Setting to `false` does not disable statistics gathering.

Reviewer: ```suggestion
    query plan and therefore improve query processing performance. Setting to `false` is not recommended and does not disable statistics gathering.
```
